### PR TITLE
Do not render extra frame while playing an audio only video file

### PR DESCRIPTION
**Pull request by ihhub**
We should not waste resources to do unnecessary work for rendering.

### CODE REVIEW

File: src/fheroes2/game/game_video.cpp [hunk 1]
```diff
@@ -40,6 +40,15 @@ namespace
         fheroes2::DrawRect( image, roi, color );
         fheroes2::DrawRect( image, fheroes2::Rect( roi.x - 1, roi.y - 1, roi.width + 2, roi.height + 2 ), color );
     }
+
+    void playAudio( const std::vector<std::vector<uint8_t>> & audioChannels )
+    {
+        for ( const std::vector<uint8_t> & audio : audioChannels ) {
+            if ( !audio.empty() ) {
+                Mixer::Play( &audio[0], static_cast<uint32_t>( audio.size() ), -1, false );
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_video.cpp:48]: The last two arguments of `Mixer::Play()` are `-1` and `false` by default, so we can just omit them.

File: src/fheroes2/game/game_video.cpp [hunk 2]
```diff
@@ -76,6 +85,17 @@ namespace Video
         if ( video.frameCount() < 1 ) // nothing to show
             return 0;
 
+        const std::vector<std::vector<uint8_t>> & audioChannels = video.getAudioChannels();
+        const bool hasAudio = Audio::isValid() && !audioChannels.empty();
+        if ( action == VideoAction::IGNORE_VIDEO ) {
+            // Since no video is rendered play audio if available.
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_video.cpp:91]: Just curious - what is the purpose of playing the soundtrack from the video while we are not playing the video itself?

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub I just left one optional suggestion and one question, please take a look when it's convenient for you.

File: src/fheroes2/game/game_video.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/game/game_video.cpp:91]: We have two problems:
- there are video files that do not have any reasonable video content
- there are video files that suppose to be played in tandem with other video sequences (like glowing crystal in campaigns) but I haven't implemented the proper way to do it

**Comment by ihhub** [comment]
Hi @oleg-derevenetz , I addressed your given comment. Please take a look.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**